src/utilities/createRouter.ts

```diff
@@ -59,8 +64,22 @@ export function createRouter<T extends Routes>(routes: T, options: RouterOptions
     Object.assign(route, newRoute)
   }
 
-  const push: RouterPush = async (url, options) => {
-    await navigation.update(url, options)
+  async function push(url: string, options?: RouterPushOptions): Promise<void>
+  async function push(route: { name: T[number]['name'] }, params?: Record<string, unknown>, options?: RouterPushOptions): Promise<void>
+  async function push(urlOrRoute: { name: T[number]['name'] } | string, optionsOrParams: Record<string, any> = {}, possiblyOptions: RouterPushOptions = {}): Promise<void> {
+    if (typeof urlOrRoute !== 'string') {
+      const route = resolved.find(({ name }) => name === urlOrRoute.name)
+
+      if (!route) {
+        throw `No route found with name "${urlOrRoute.name}"`
+      }
+
+      const url = assembleUrl(route, optionsOrParams)
+
+      return push(url, possiblyOptions)
+    }
+
+    await navigation.update(urlOrRoute, optionsOrParams)
```

Comment:
I wonder if we can abstract the implementation of this into another file. createRouter is so small and easy to read. Be nice to keep logic out of it. Just a thought. Not an action item.